repo_resource/common.py

```diff
@@ -185,34 +277,33 @@ def sync(self, version: Version = None, jobs: int = 0):
                 if jobs > 0:
                     repo_cmd.append('--jobs={}'.format(jobs))
 
-                if version is None:
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    tmp_manifest = os.path.join(tmpdir, 'manifest_tmp')
+                    version.to_file(tmp_manifest)
+                    repo_cmd.append(
+                        '--manifest-name={}'.format(tmp_manifest))
                     repo._Main(repo_cmd)
-                else:
-                    with tempfile.TemporaryDirectory() as tmpdir:
-                        tmp_manifest = os.path.join(tmpdir, 'manifest_tmp')
-                        version.to_file(tmp_manifest)
-                        repo_cmd.append(
-                            '--manifest-name={}'.format(tmp_manifest))
-                        repo._Main(repo_cmd)
+                if os.listdir(self.__workdir) == []:
+                    raise Exception('Sync failed. Is manifest correct?')
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    tmp_manifest = os.path.join(tmpdir, 'manifest_snapshot')
+                    self.__manifest_out(tmp_manifest)
+                    self.__version = Version.from_file(tmp_manifest)
         except Exception as e:
+            sys.exit(1)
```

Comment:
Why is this added? shouldn't we restore the old pwd if an exception happens ?

Reply:
Removed. Tests have been changed to match expected errors